django/db/models/functions/datetime.py

```diff
@@ -248,7 +248,7 @@ def output_field(self):
         return TimeField()
 
     def as_sql(self, compiler, connection):
-        # Cast to date rather than truncate to date.
+        # Cast to time rather than truncate to time.
         lhs, lhs_params = compiler.compile(self.lhs)
         tzname = timezone.get_current_timezone_name() if settings.USE_TZ else None
         sql, tz_params = connection.ops.datetime_cast_time_sql(lhs, tzname)
```

Comment:
I think we should only perform this step if `isinstance(self.lhs.output_field, DateTimeField)`.